homeassistant/components/smarttub/climate.py

```diff
@@ -0,0 +1,119 @@
+"""Platform for climate integration."""
+import logging
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    HVAC_MODE_HEAT,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+from homeassistant.util import slugify
+from homeassistant.util.temperature import convert as convert_temperature
+
+from .const import DEFAULT_MAX_TEMP, DEFAULT_MIN_TEMP, DOMAIN, SMARTTUB_CONTROLLER
+from .entity import SmartTubEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up climate entity for the thermostat in the tub."""
+
+    controller = hass.data[DOMAIN][entry.unique_id][SMARTTUB_CONTROLLER]
```

Comment:
```suggestion
    controller = hass.data[DOMAIN][entry.entry_id][SMARTTUB_CONTROLLER]
```

We usually store by `entry_id` instead of `unique_id` as unique ids can be migrated whereas `entry_id` is guaranteed to be stable.